### PR TITLE
Adding flags to datamodules

### LINKED ISSUE #245: pin_memory should be true only if gpus are specified
Hey folks,

First of all, thank you for this amazing package. It has saved many hours of coding :) 

I was using for the first time the pl data modules when I hit the following line
[cifar10_datamodule.py#L132](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/c2cdad7c12e4d33ed39db808879706c608c1c7/pl_bolts/datamodules/cifar10_datamodule.py#L132). The problem is: when I am training/evaluating/testing on CPU, the data module try to pin the data and try to reserve a batch size memory amount in gpu. Nothing goes wrong if I have GPU with enough memory ran, but, when I try to run in CPU mode with all my GPUs in use I get the following error:

```
RuntimeError: Caught RuntimeError in pin memory thread for device 0.
Original Traceback (most recent call last):
  File "/lib/python3.7/site-packages/torch/utils/data/_utils/pin_memory.py", line 31, in _pin_memory_loop
    data = pin_memory(data)
  File "/lib/python3.7/site-packages/torch/utils/data/_utils/pin_memory.py", line 55, in pin_memory
    return [pin_memory(sample) for sample in data]
  File "lib/python3.7/site-packages/torch/utils/data/_utils/pin_memory.py", line 55, in <listcomp>
    return [pin_memory(sample) for sample in data]
  File "lib/python3.7/site-packages/torch/utils/data/_utils/pin_memory.py", line 47, in pin_memory
    return data.pin_memory()
RuntimeError: cuda runtime error (2) : out of memory at /opt/conda/conda-bld/pytorch_1595629427478/work/aten/src/THC/THCCachingHostAllocator.cpp:278
```

The fix is simple and we can borrow the code from the pl core implementation as in [pytorch_lightning/accelerators/accelerator_connector.py#L90](https://github.com/PyTorchLightning/pytorch-lightning/blob/0045119b3fd4532bf8edf17dcbaeba9bc66ba25b/pytorch_lightning/accelerators/accelerator_connector.py#L90)

Bests.

### PR DESCRIPTION
**Pull request by briankosw**
## What does this PR do?

Adds the flags `shuffle`, `drop_last`, and `pin_memory` to datamodules.

Fixes #245

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [] Did you write any new necessary tests?
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/CHANGELOG.md)?

## PR review
 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by briankosw** [comment]
@nateraw would love to get your feedback on this PR. Specifically, I wanted some feedback on whether the flags for validation and test dataloaders should be parametrized the same way as the train dataloader. For example, previously `shuffle` was set to `True for train and `False` for validation and test. Should they both be set as the newly added flag?

File: pl_bolts/datamodules/kitti_datamodule.py [hunk 1]
```diff
@@ -21,6 +21,7 @@ def __init__(
             num_workers: int = 16,
             batch_size: int = 32,
             seed: int = 42,
+            shuffle: bool = False,
```
Comment (akihironitta) [inline pl_bolts/datamodules/kitti_datamodule.py:24]: Shall we have `drop_last` and `pin_memory` here like other datamodules?

**Review by akihironitta — CHANGES_REQUESTED** [review]
@briankosw Thank you for your contribution as always! It seems the doc tests failed. Would you mind having a look?

**Comment by akihironitta** [comment]
@briankosw mind resolving the conflicts, too?

File: pl_bolts/datamodules/kitti_datamodule.py (hunk 1 above)
Comment (briankosw) [reply pl_bolts/datamodules/kitti_datamodule.py:24]: Yeah I'm on it!

**Comment by briankosw** [comment]
Hey @akihironitta,
1. Do you think `shuffle=False` should be hardcoded in there for val and test loaders?
2. Do you find the default values that I've given for `shuffle`, `pin_memory`, and `drop_last` to be reasonable?

**Comment by akihironitta** [comment]
> 1. Do you think `shuffle=False` should be hardcoded in there for val and test loaders?

Since they are not used for training, I guess hardcoding `shuffle=False` sounds fine...
@Borda What do you think about the above?

> 2. Do you find the default values that I've given for `shuffle`, `pin_memory`, and `drop_last` to be reasonable?

Yes, it looks good to me as is :] As `DataLoader` uses `False` by default, let's keep them all `False`. https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
@akihironitta mind check if your comments were resolved, otherwise it lgtm

**Review by akihironitta — APPROVED** [review]
@briankosw LGTM! Thank you :]

**Comment by briankosw** [comment]
> > 1. Do you think `shuffle=False` should be hardcoded in there for val and test loaders?
> 
> Since they are not used for training, I guess hardcoding `shuffle=False` sounds fine...

Sounds good! I went ahead and hardcoded `shuffle=False` for validation and testing. Do I need to add anything to the documentation or the changelog for this PR?

**Pull request merged**